packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -153,7 +175,7 @@ library LibProposing {
         // Ethereum block, we choose to introduce a salt to this random
         // number as the L2 mixHash.
         meta_.difficulty =
-            keccak256(abi.encodePacked(block.prevrandao, local.b.numBlocks, block.number));
+            keccak256(abi.encodePacked(local.params.anchorBlockId, local.b.numBlocks, block.number));
```

Comment:
This reverts back to the dependency on block.number that I fixed in my PR: https://github.com/taikoxyz/taiko-mono/pull/17654#issuecomment-2189462000. Using `blockhash(params.l1StateBlockNumber)` also indirectly does depend on a randao instead of just using `l1StateBlockNumber`.

Reply:
I update the code but it is still different from your original change. Please take another look.